### PR TITLE
Add tests for EmrServerlessJobSensor and EmrServerlessApplicationSensor

### PR DESCRIPTION
**Pull request by mateuslatrova**
Related: #35442 

This PR adds unit tests for both the **EmrServerlessJobSensor** and the **EmrServerlessApplicationSensor** from the Airflow AWS provider package.

### CODE REVIEW

**Comment by mateuslatrova** [comment]
Obs.: still working on the tests for **EmrServerlessApplicationSensor**

**Review by dirrao — COMMENTED** [review]
The serverless tests are failing. Can you check and fix them?

**Comment by mateuslatrova** [comment]
> The serverless tests are failing. Can you check and fix them?

Sure! I will soon fix them and also add the tests for EmrServerlessApplicationSensor.

**Comment by mateuslatrova** [comment]
Hi @dirrao !

Just fixed the tests for EmrServelessJobSensor, whenever you can take a look.

I decided to do a separate PR with the tests for the EmrServerlessApplicationSensor.

Thanks in advance!

File: tests/providers/amazon/aws/sensors/test_emr_serverless.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+        self.run_test(get_job_run_return_value={"jobRun": {"state": "PENDING"}})
+
+    def test_when_state_is_running_poke_should_return_false(self):
+        self.run_test(get_job_run_return_value={"jobRun": {"state": "RUNNING"}})
+
+    def test_when_state_is_scheduled_poke_should_return_false(self):
+        self.run_test(get_job_run_return_value={"jobRun": {"state": "SCHEDULED"}})
+
+    def test_when_state_is_submitted_poke_should_return_false(self):
+        self.run_test(get_job_run_return_value={"jobRun": {"state": "SUBMITTED"}})
```
Comment (vincbeck) [inline tests/providers/amazon/aws/sensors/test_emr_serverless.py]: These tests could be one test using [parametrize](https://docs.pytest.org/en/7.1.x/how-to/parametrize.html)

File: tests/providers/amazon/aws/sensors/test_emr_serverless.py [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+
+class TestPokeReturnsTrue(TestEmrServerlessJobSensor):
+    def test_when_state_is_success_poke_should_return_true(self):
+        self.set_get_job_run_return_value({"jobRun": {"state": "SUCCESS"}})
+        assert self.sensor.poke(None)
+        self.assert_get_job_run_was_called_once_with_app_and_run_id()
```
Comment (vincbeck) [inline tests/providers/amazon/aws/sensors/test_emr_serverless.py]: You could include this test as part of the test above using [parametrize](https://docs.pytest.org/en/7.1.x/how-to/parametrize.html)

File: tests/providers/amazon/aws/sensors/test_emr_serverless.py (hunk 1 above)
Comment (mateuslatrova) [reply tests/providers/amazon/aws/sensors/test_emr_serverless.py]: Oh, nice. I am more familiar with unittest features and never used `parametrize` before. But I'll take a look at it and do these changes until tomorrow. Thanks.

File: tests/providers/amazon/aws/sensors/test_emr_serverless.py (hunk 2 above)
Comment (mateuslatrova) [reply tests/providers/amazon/aws/sensors/test_emr_serverless.py]: Same answer as above.

**Comment by mateuslatrova** [comment]
Hi @vincbeck ! Thanks for your suggestion, the code became much more cleaner. I have just uploaded the changes, whenever you can review it again. Thanks!

**Comment by mateuslatrova** [comment]
Since this MR is still not merged, and both sensors (EmrServerlessJobSensor and EmrServerlessApplicationSensor) are very related, I have just added here a commit with the tests for EmrServerlessApplicationSensor.

**Review by vincbeck — APPROVED** [review]

**Comment by vincbeck** [comment]
Love it! Thanks!

**Pull request merged**